akka-docs/rst/java/code/docs/http/javadsl/server/directives/BasicDirectivesExamplesTest.java

```diff
@@ -0,0 +1,788 @@
+      final Source<ByteString, Object> source =
+        FileIO.fromPath(Paths.get("example.json")).mapMaterializedValue(completionStage -> (Object) completionStage);
+      return complete(
+        HttpResponse.create().withEntity(HttpEntities.create(ContentTypes.APPLICATION_JSON, source))
+      );
+    });
+
+    final Route route = get(() ->
+      route(
+        pathPrefix("special", () ->
+          withSettings(special, () -> sample) // `special` file-io-dispatcher will be used to read the file
+        ),
+        sample // default file-io-dispatcher will be used to read the file
+      )
+    );
+
+    // tests:
+    testRoute(route).run(HttpRequest.GET("/special/sample"))
+      .assertEntity("{}");
+    testRoute(route).run(HttpRequest.GET("/sample"))
+      .assertEntity("{}");
+    //#withSettings
+  }
+
+  @Test
+  public void testMapResponse() {
+    //#mapResponse
+    final Route route = mapResponse(
+      response -> response.withStatus(StatusCodes.BAD_GATEWAY),
+      () -> complete("abc")
+    );
+
+    // tests:
+    testRoute(route).run(HttpRequest.GET("/abcdef?ghi=12"))
+      .assertStatusCode(StatusCodes.BAD_GATEWAY);
+    //#mapResponse
+  }
+
+  class ApiRoute {
+
+    private ActorSystem _system = null;
+
+    private LoggingAdapter log = null;
+
+    private HttpEntity nullJsonEntity = HttpEntities.create(ContentTypes.APPLICATION_JSON, "{}");
+
+    public ApiRoute(ActorSystem system) {
+      _system = system;
+      log = Logging.getLogger(system, "ApiRoutes");
+    }
+
+    private HttpResponse nonSuccessToEmptyJsonEntity(HttpResponse response) {
+      if (response.status().isSuccess()) {
+        return response;
+      } else {
+        log.warning("Dropping response entity since response status code was: " + response.status());
+        return response.withEntity((ResponseEntity) nullJsonEntity);
+      }
+    }
+
+    /** Wrapper for all of our JSON API routes */
+    private Route apiRoute(Supplier<Route> innerRoutes) {
+      return mapResponse(this::nonSuccessToEmptyJsonEntity, innerRoutes);
+    }
+  }
+
+  @Test
+  public void testMapResponseAdvanced() {
+    //#mapResponse-advanced
+    final ApiRoute api = new ApiRoute(system());
+
+    final Route route = api.apiRoute(() ->
+      get(() -> complete(StatusCodes.INTERNAL_SERVER_ERROR))
+    );
+
+    // tests:
+    testRoute(route).run(HttpRequest.GET("/"))
+      .assertEntity("{}");
+    //#mapResponse-advanced
+  }
+
+  @Test
+  public void testMapRouteResult() {
+    //#mapRouteResult
+    // this directive is a joke, don't do that :-)
+    final Route route = mapRouteResult(r -> {
+      if (r instanceof akka.http.scaladsl.server.RouteResult.Complete) {
+        final HttpResponse response = ((akka.http.scaladsl.server.RouteResult.Complete) r).getResponse();
+        return new akka.http.scaladsl.server.RouteResult.Complete(
+          (akka.http.scaladsl.model.HttpResponse) response.withStatus(200)
```

Comment:
As noted in the "general" comment, no scaladsl and no fully qualified classes here, should make it much much cleaner and easy to read.